src/ieee80211_station.erl

```diff
@@ -1736,21 +1736,22 @@ handle_session_timer_ev({_, Level, _} = Ev, {Interval, _, _Opts} = Timer,
 
 % FIXME Atoms are allowed for test cases. Test cases should, however,
 % use DTLS and not plain UDP.
-add_location(ACAtom, Map) when is_atom(ACAtom) ->
-    ?LOG(debug, "Getting location through atom: ~p", [ACAtom]),
-    add_location(atom_to_binary(ACAtom, utf8), Map);
-add_location(AC, Map) when is_binary(AC) ->
+add_location(AC, Map) when is_pid(AC) ->
     ?LOG(debug, "Getting location through: ~p", [AC]),
     try capwap_ac:get_info(AC) of
 	{ok, #{id := Name}} ->
 	    ?LOG(debug, "Getting location for name: ~p", [Name]),
-	    AVPName = capwap_config:get(ac, location_avp, undefined),
-	    case capwap_loc_provider:get_loc(Name, false) of
+	    BinName = case Name of _ when is_binary(Name) -> Name;
+				   _ when is_atom(Name) -> atom_to_binary(Name, utf8);
+				   _ -> <<"undefined">> end,
+	    ?LOG(debug, "Getting location for name: ~p", [{Name, BinName}]),
+	    SessionAttr = 'IM-LI-Location',
+	    case capwap_loc_provider:get_loc(BinName, false) of
 		{location, LatVal, LongVal} ->
 		    ?LOG(debug, "Successful location: ~p", [{LatVal, LongVal}]),
-		    Map#{AVPName => iolist_to_binary(io_lib:format("Lat:~s;Lon:~s", [LatVal, LongVal]))};
+		    Map#{SessionAttr => iolist_to_binary(io_lib:format("Lat:~s;Lon:~s", [LatVal, LongVal]))};
```

Comment:
stop this nonsense. ergw_aaa already contains support for handling the vendor specific attribute. Put the values into the `CAPWAP-GPS-Latitude` and `CAPWAP-GPS-Longitude` session attributes and let ergw_aaa handle the rest.

Reply:
There is no need to perform additional parsing and composing that will be done by someone else above in the chain of services. If the user has separate GPS coordinates that need to be assembled, this is fine. A location provider provides now a location string already assembled.

Reply:
And it should absolutely not do that. It is the task of the ergw_aaa library to format internal data types for export on the various APIs. The internal representation should be independent of the external APIs.

What you are doing here is tailoring the internal location provider for the suite for a single external API. That is not the way the API where intended to used or are currently use.